File: WaterManager/src/main/java/pissir/watermanager/model/user/GestoreIdrico.java
package pissir.watermanager.model.user;


import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import pissir.watermanager.model.item.BacinoIdrico;

/**
 * @author Almasio Luca
 * @author Borova Dritan
 * @author Gattico Alessandro
 */
@Setter
@Getter
@NoArgsConstructor
public class GestoreIdrico extends UserProfile {
	
	private BacinoIdrico bacinoIdrico;
	
	
	public GestoreIdrico(int id, String nome, String cognome, String username, String mail, String password,
						 boolean enabled) {
		super(id, nome, cognome, username, mail, password, UserRole.GESTOREIDRICO, enabled);
		this.bacinoIdrico = null;
	}
	
}

File: WaterManagerUI/Model/User/GestoreIdrico.cs
using System;
using WaterManagerUI.Model.Item;

namespace UserInterfaceWaterManager.Model.User
{
    public class GestoreIdrico : UserProfile
    {
        public BacinoIdrico bacinoIdrico { get; set; }
        public HashSet<RisorsaIdrica> risorse { get; set; }

        public GestoreIdrico(int id, String nome, String cognome, String username, String mail, String password,
                             BacinoIdrico bacinoIdrico) : base(id, nome, cognome, username, mail, password, UserRole.GESTOREIDRICO)
        {
            this.bacinoIdrico = bacinoIdrico;
            this.risorse = new HashSet<RisorsaIdrica>();
        }

    }
}

